src/system/modifier-data.ts

```diff
@@ -13,7 +13,7 @@ export default class ModifierData {
   public className: string;
 
   constructor(source: PersistentModifier | any, player: boolean) {
-    const sourceModifier = source instanceof PersistentModifier ? (source as PersistentModifier) : null;
+    const sourceModifier = source.isPersistentModifier() ? source : null;
```

Comment:
```suggestion
    const sourceModifier = source.isPersistentModifier && source.isPersistentModifier() ? source : null;
```
(`sourceModifier` can also be loaded modifier data here.)